test/demo/contextChannels.test.ts

```diff
@@ -0,0 +1,270 @@
+        }
+
+        return identity;
+    }));
+
+    return result;
+}
+
+describe('When broadcasting on global channel', () => {
+    it('Context is received by global windows only', async () => {
+        const [globalWindow, blueWindow] = await setupWindows(undefined, 'blue');
+
+        const globalListener = await fdc3Remote.addContextListener(globalWindow);
+        const blueListener = await fdc3Remote.addContextListener(blueWindow);
+
+        // Broadcast our context on the global channel
+        await fdc3Remote.broadcast(testManagerIdentity, testContext);
+
+        // Check the global window received our test context
+        const globalContexts = await globalListener.getReceivedContexts();
+        expect(globalContexts.length).toBe(1);
+        expect(globalContexts[0]).toEqual(testContext);
+
+        // Check the blue window received no context
+        const blueContexts = await blueListener.getReceivedContexts();
+        expect(blueContexts.length).toBe(0);
+    });
+
+    it('Context is received by window that has left and rejoined global channel', async () => {
+        const [channelChangingWindow] = await setupWindows('blue');
+
+        // Change the channel of our window
+        await fdc3Remote.joinChannel(channelChangingWindow, 'global');
+
+        const channelChangingListener = await fdc3Remote.addContextListener(channelChangingWindow);
+
+        // Broadcast our context on the global channel
+        await fdc3Remote.broadcast(testManagerIdentity, testContext);
+
+        // Check our window received our test context
+        const receivedContexts = await channelChangingListener.getReceivedContexts();
+        expect(receivedContexts.length).toBe(1);
+        expect(receivedContexts[0]).toEqual(testContext);
+    });
+});
+
+describe('When broadcasting on a user channel', () => {
+    it('Context is received by windows on that user channel only', async () => {
+        const [red1Window, red2Window, globalWindow, blueWindow] = await setupWindows('red', 'red', 'global', 'blue');
+
+        const redListener = await fdc3Remote.addContextListener(red2Window);
+        const globalListener = await fdc3Remote.addContextListener(globalWindow);
+        const blueListener = await fdc3Remote.addContextListener(blueWindow);
+
+        // Broadcast our context on the red channel
+        await fdc3Remote.broadcast(red1Window, testContext);
+
+        // Check our red window received our test context
+        const redReceivedContexts = await redListener.getReceivedContexts();
+        expect(redReceivedContexts.length).toBe(1);
+        expect(redReceivedContexts[0]).toEqual(testContext);
+
+        const globalReceivedContexts = await globalListener.getReceivedContexts();
+        expect(globalReceivedContexts.length).toBe(0);
+
+        // Check our blue window received no context
+        const blueReceivedContexts = await blueListener.getReceivedContexts();
+        expect(blueReceivedContexts.length).toBe(0);
```

Comment:
`expect(blueReceivedContexts).toHaveLength(0);`?

Check is equivalent, but should give more useful error messages if it fails(?).